### DISCUSSION TITLE
Can you use the node REPL with Sveltekit? What about run utility scripts?

### DISCUSSION THREAD

**Discussion by Glench** [sveltejs/kit#8753] [Q&A]
I'm wondering if it's possible to open a node repl and have imports and all that work out with SvelteKit? Quite often in production I have to do one-off things that require either to be in a repl or to call a script that lives outside the web server but still needs access to things like `$lib/db` or `$app`, or `$env/static/private`. In my previous app I just used `require` and relative paths, but that doesn't seem to work with SvelteKit.

**Comment by wvhulle** [comment]
You cannot use $app in node. It only exists in a vite dev or build. For $lib you need a special node loader. Ts-paths-esm-loader is one. It translates the lib to imports node can understand

**Reply by Glench** [reply]
Thanks for some of the info! However, maybe I should change the question to "I *need* access to `$app` — how do I make that happen?" For example, my database library needs to have access to $app/environment to figure out which db to connect to. I also need to use the db library outside of SvelteKit, e.g. in repls and scripts. What are some ways to get this to work?

**Comment by geoffrich** [comment]
Not sure about all the SK-specific imports, but `$app/environment` uses [esm-env](https://github.com/benmccann/esm-env) under the hood, so you could use that directly.

https://github.com/sveltejs/kit/blob/4ae8229276fd851f1d0f145b6ffd0c3b23e94fea/packages/kit/src/runtime/app/environment.js#L1-L11

**Reply by Glench** [reply]
Oh good call! That looks like it should work for `$app/environment.dev`. I guess I should also mention that I want to use `$env` as well outside of SvelteKit. I edited my original post to mention that.
